front/app/modules/commercial/segment/index.tsx

```diff
@@ -62,25 +32,42 @@ const integrations = (user: IUser | null) => {
   return output;
 };
 
+let isSegmentEnabled = false;
+
 const configuration: ModuleConfiguration = {
   beforeMountApplication: () => {
-    initializeFor('segment').subscribe(() => {
-      const code = snippet.min({
-        host: 'cdn.segment.com',
-        load: true,
-        page: false,
-        apiKey: CL_SEGMENT_API_KEY,
-      });
-
-      // eslint-disable-next-line no-eval
-      eval(code);
-    });
+    if (!CL_SEGMENT_API_KEY) return;
 
     combineLatest([
       currentAppConfigurationStream().observable,
-      bufferUntilInitialized('segment', authUserStream().observable),
+      authUserStream().observable,
     ]).subscribe(([tenant, user]) => {
+      const segmentFeatureFlag = tenant.data.attributes.settings.segment;
+      isSegmentEnabled = Boolean(
+        // Feature flag is in place
+        segmentFeatureFlag?.allowed &&
+          segmentFeatureFlag?.enabled &&
+          // User is admin or moderator
+          !isNilOrError(user) &&
+          (isAdmin(user) || isModerator(user) || isProjectModerator(user))
```

Comment:
Every user with a role higher than user is considered a moderator, so it does include folder moderators and the rest of the checks are redundant, as well.

Reply:
Those functions are confusingly named then, I feel, I would expect them to be false in case it's not an exact role match. If it's easy to fix, would be nice to make it more logical.

 I think we have/had a highestRole function for such cases?

Reply:
I agree! We should revisit this once we start working on permissions 👍